src/fs/rock/RockRebuild.h

```diff
@@ -24,20 +27,47 @@ class LoadingParts;
 
 /// \ingroup Rock
 /// manages store rebuild process: loading meta information from db on disk
-class Rebuild: public AsyncJob
+class Rebuild: public AsyncJob, private IndependentRunner
 {
     CBDATA_CHILD(Rebuild);
 
 public:
-    Rebuild(SwapDir *dir);
-    virtual ~Rebuild() override;
+    /// cache_dir indexing statistics shared across same-kid process restarts
+    class Stats
+    {
+    public:
+        static SBuf Path(const char *dirPath);
+        static Ipc::Mem::Owner<Stats> *Init(const SwapDir &);
+
+        static size_t SharedMemorySize() { return sizeof(Stats); }
+        size_t sharedMemorySize() const { return SharedMemorySize(); }
+
+        /// whether the rebuild is finished already
+        bool completed(const SwapDir &) const;
+
+        StoreRebuildData counts;
+    };
+
+    /// starts indexing the given cache_dir if that indexing is necessary
+    /// \returns whether the indexing was necessary (and, hence, started)
+    static bool Start(SwapDir &dir);
 
 protected:
+    /// whether the current kid is responsible for rebuilding this db file
+    static bool IsResponsible(const SwapDir &);
+
+    Rebuild(SwapDir *dir, const Ipc::Mem::Pointer<Stats> &);
+    virtual ~Rebuild() override;
+
+    /* Registered Runner API */
+    virtual void startShutdown() override;
+
     /* AsyncJob API */
     virtual void start() override;
     virtual bool doneAll() const override;
     virtual void swanSong() override;
 
+    bool loadedAndValidated() const { return doneLoading() && doneValidating(); }
```

Comment:
Let's remove this used-once conjunction method, undoing the non-change in its caller.

Reply:
Done.